examples/knative/python/kn-py-vro/README.md

```diff
@@ -0,0 +1,272 @@
+# kn-py-vro <!-- omit in toc -->
+- [Overview](#overview)
+- [Step 1 - Build with `pack`](#step-1---build-with-pack)
+- [Step 2 - Test](#step-2---test)
+- [Step 3 - Deploy](#step-3---deploy)
+  - [Prerequisites](#prerequisites)
+  - [Push container image to registry](#push-container-image-to-registry)
+  - [Configuration](#configuration)
+  - [Deploy](#deploy)
+- [Step 4 - Undeploy](#step-4---undeploy)
+- [vRO Workflow](#vro-workflow)
+- [Configuration](#configuration-1)
+  - [vRO Workflow ID](#vro-workflow-id)
+  - [OAUTH via VMWare Cloud Services Console - for CEXP](#oauth-via-vmware-cloud-services-console---for-cexp)
+
+# Overview
+This is an example Python function with `Flask` REST API running in Knative to trigger a vRealize Orchestrator (vRO) workflow using the parameters of an event submitted by the [VEBA](https://vmweventbroker.io/) (VMware Event Broker) services.
+
+The Cloud Event parameters are converted in to native vRO datatypes wherever possible, including vCenter SDK Managed Objects, and passed as inputs to a defined vRO Workflow.
+
+The function code not only demonstrates a number of techniques (such as creating persistent OAUTH sessions, and the vRO workflow API), but is generic enough to be usable in a multitude of scenario's out-of-the-box.
+
+> **Note:** Only Cloud Event payload ("*data*") in JSON format is supported.
+
+
+# Step 1 - Build with `pack`
+
+> **Note:** If you do not plan to edit the function behavior, you can go directly to [Step 3 - Deploy](#step-3---deploy) part of this document.
+
+[Buildpacks](https://buildpacks.io) are used to create the container image.
+
+Requirements:
+
+- `pack` (see: https://buildpacks.io/docs/app-developer-guide/)
+- Docker
+
+```bash
+export IMAGE=<docker-username>/kn-py-vro
+pack build --builder gcr.io/buildpacks/builder:v1 ${IMAGE}
+```
+
+
+# Step 2 - Test
+
+1. Edit the content of `kn-py-vro_secret.json` to configure the following settings (see [Configuration](#configuration) below for details):
+
+    ```json
+    {
+        "SERVER": "VRO_FQDN_OR_IP",
+        "PORT": 443,
+        "AUTH_TYPE": "BASIC/OAUTH",
+        "USERNAME": "VRO_USERNAME",
+        "PASSWORD": "VRO_PASSWORD",
+        "REFRESH_TOKEN": "VRAC_ONLY",
+        "INSECURE_SSL": false,
+        "WORKFLOW_ID": "VRO_WORKFLOW_ID"
+    }
+    ```
+
+2. Run a local container based on the `${IMAGE}` image or the one you just built.
+
+    ```bash
+    docker run -e PORT=8080 -it --rm -p 8080:8080 --env VCCONFIG_SECRET="$(cat kn-py-vm-attr_secret.json)" ${IMAGE}
```

Comment:
nit: would that work on windows (`cat`)?

Reply:
"Who uses windows!?" - oh wait, me right now! I'll fix.

I based this on the kn-py-vm-attr function - we should fix a few bits in there too. Should I raise an issue?